jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -83,5 +90,44 @@ def process(self, input, *args, **kwargs):
             slit_y_range = [self.slit_y_low, self.slit_y_high]
             result = load_wcs(input_model, reference_file_names, slit_y_range)
 
+        if not (result.meta.exposure.type.lower() in IMAGING_TYPES and self.sip_approx):
```

Comment:
@cshanahan1 I think this is the wrong condition that makes sip approximation run for spectral types. We need this to evaluate to True in order to skip SIP processing. As it is written it will evaluate to False for spectral observations when `self.sip_approx=True`. If it's written as

` if result.meta.exposure.type.lower() not in IMAGING_TYPES or not self.sip_approx:`

Reply:
> @cshanahan1 I think this is the wrong condition that makes sip approximation run for spectral types. We need this to evaluate to True in order to skip SIP processing. As it is written it will evaluate to False for spectral observations when `self.sip_approx=True`. If it's written as
> 
> ` if result.meta.exposure.type.lower() not in IMAGING_TYPES or not self.sip_approx:`

I disagree. If image type's is not in `IMAGING_TYPES` then  `result.meta.exposure.type.lower() in IMAGING_TYPES` is `False` and combined with the `not` in front => `True` => `return result` without fitting SIP.

Reply:
Am I missing something?

Reply:
I agree :-) with your disgreement